featuretools/feature_base/feature_base.py

```diff
@@ -462,13 +463,34 @@ def generate_name(self):
         return self.primitive.generate_name(base_feature_names=[bf.get_name() for bf in self.base_features])
 
 
+class GroupByTransformFeature(TransformFeature):
+    def __init__(self, base_features, primitive, groupby):
+        if not isinstance(groupby, FeatureBase):
+            groupby = IdentityFeature(groupby)
+        assert issubclass(groupby.variable_type, Discrete)
+        self.groupby = groupby
+        super(GroupByTransformFeature, self).__init__(base_features,
+                                                      primitive=primitive)
+        self.base_features.append(groupby)
```

Comment:
why not add this to `base_features` before calling the super method?

Reply:
also, not sure it is worth subclassing `TransformFeature`. Doesn't seem like they actually share much.